### PR TITLE
[mkldnn-v1.0] Add MKL-DNN activation

### PR DESCRIPTION
**Pull request by rongzha1**
## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

@ciyongch @ZhennanQin @TaoLv @PatricZhao

### CODE REVIEW

**Comment by pengzhao-intel** [comment]
@TaoLv @ZhennanQin @ciyongch @xinyu-intel please take a review?

File: src/operator/nn/mkldnn/mkldnn_act-inl.h [hunk 1]
```diff
@@ -67,8 +63,28 @@ MKLDNNActForward &GetActForward(const ActivationParam& param,
 void MKLDNNActivationForward(const nnvm::NodeAttrs& attrs, const OpContext &ctx,
                              const NDArray &in_data, const OpReqType &req,
                              const NDArray &out_data);
+
+mkldnn::eltwise_backward::primitive_desc GetActBwdDescImpl(
+    const ActivationParam &param, const mkldnn::memory &input_mem,
+    const mkldnn::memory &diff_dst_memory);
+
+class MKLDNNActBackward {
+  std::shared_ptr<mkldnn::eltwise_backward> bwd;
```
Comment (ciyongch) [inline src/operator/nn/mkldnn/mkldnn_act-inl.h]: Make this as a private member? As there's an API `GetBwd()` to retrieve it.

Comment (rongzha1) [reply src/operator/nn/mkldnn/mkldnn_act-inl.h]: Thanks. Will change it.

**Comment by pengzhao-intel** [comment]
@TaoLv please merge the related 1.0 PR when the branch is good.

**Pull request merged**